lib/thing_importer.rb

```diff
@@ -1,52 +1,70 @@
 require 'open-uri'
-
+require 'uri'
+require 'net/http'
+require 'json'
 # class for importing things from CSV datasource
-# is currently very specific to drains from DataSF
+# is currently very specific to drains from Grand River Basin
 #
 # Dataset:
 # https://data.sfgov.org/City-Infrastructure/Stormwater-inlets-drains-and-catch-basins/jtgq-b7c5
+# Data Flow (TABLES):
+#   csv -> temp_thing_import -> things
+# App Flow
+# -> ThingImporter
 class ThingImporter
   class << self
     def load(source_url)
+      # puts 'Downloading Things '
+      # puts 'DW_USER: ' + ENV['DW_USER']
+      # puts 'OPEN_SOURCE: ' + ENV['OPEN_SOURCE']
+
       Rails.logger.info('Downloading Things... ... ...')
 
       ActiveRecord::Base.transaction do
+
         import_temp_things(source_url)
 
         deleted_things_with_adoptee, deleted_things_no_adoptee = delete_non_existing_things
+
         created_things = upsert_things
 
         ThingMailer.thing_update_report(deleted_things_with_adoptee, deleted_things_no_adoptee, created_things).deliver_now
+
       end
+
     end
 
     def integer?(string)
       return true if string =~ /\A\d+\Z/
-
       false
     end
 
-    def normalize_thing(csv_thing)
-      (lat, lng) = csv_thing['Location'].delete('()').split(',').map(&:strip)
-      {
-        city_id: csv_thing['PUC_Maximo_Asset_ID'].gsub!('N-', ''),
-        lat: lat,
-        lng: lng,
-        type: csv_thing['Drain_Type'],
-        system_use_code: csv_thing['System_Use_Code'],
+    def normalize_thing(json_thing)
+      norm = {
+        city_id: json_thing['dr_facility_id'],
+        lat: json_thing['dr_lat'],
+        lng: json_thing['dr_lon'],
+        type: json_thing['dr_type'],
+        system_use_code: json_thing['dr_subwatershed'],
       }
+
+      return norm
     end
 
     def invalid_thing(thing)
-      false unless ['Storm Water Inlet Drain', 'Catch Basin Drain'].include?(thing[:type])
-      false unless integer?(thing[:city_id])
-      true
+      # puts 'invalid_thing 1'
+      rc = false
+      false unless ['Storm Water Inlet Drain', 'Catch Basin Drain'].include?(thing["type"])
+      false unless integer?(thing["city_id"])
```

Comment:
These 2 lines aren't doing anything either. I think they are supposed to be `return false unless...`

Ruby has implicit return, but that only works on the last branch of any block. Below there is an explicit `return true`, which could be re-written as just `true` -- though it's fine to explicitly return too.

Reply:
Prefer explicit return.